catlin/pkg/validator/kind_validator.go

```diff
@@ -0,0 +1,30 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"github.com/tektoncd/plumbing/catlin/pkg/app"
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+)
+
+func NewKindValidator(cli app.CLI, r *parser.Resource) Validator {
+
+	switch r.Kind {
+	case "Task":
+		return &TaskValidator{res: r, log: cli.Logger().Sugar()}
+	}
+
+	return nil
```

Comment:
this should return an error `UnknownKindError` instead of nil which can be checked by the caller before adding to the list of `validators`